### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by olorin**
Wrote some additional tests while debugging another issue - didn't turn out to be the problem, but figured I may as well commit them.

### CODE REVIEW

File: x-conduit/test/Test/X/Data/Conduit/Binary.hs [hunk 1]
```diff
@@ -37,14 +38,26 @@ prop_slurp =
     forAll (choose (100, 4096) :: Gen Int) $ \buffer ->
       forAll (choose (0, length bs)) $ \offset ->
         forAll (choose (0, (length bs) - offset)) $ \chunk -> testIO $ do
-          withSystemTempDirectory "conduit" $ \p -> do
+          withTempDirectory "." "conduit" $ \p -> do
```
Comment (nhibberd) [inline x-conduit/test/Test/X/Data/Conduit/Binary.hs]: What's the reason for this?

Comment (nhibberd) [inline x-conduit/test/Test/X/Data/Conduit/Binary.hs]: Curious why not just set TMPDIR in the build scripts I guess

Comment (olorin) [inline x-conduit/test/Test/X/Data/Conduit/Binary.hs]: See the "robust builds" guide in the engineering repo, makes it easier to deal with tmpdir space issues.

Comment (olorin) [inline x-conduit/test/Test/X/Data/Conduit/Binary.hs]: Could do that too. Probably a better option come to think of it.

**Comment by tmcgilchrist** [comment]
More tests :1234:

**Pull request merged**